apps/prairielearn/src/sync/course-db.ts

```diff
@@ -1227,11 +1227,34 @@ async function validateAssessment(
               'Cannot specify "maxPoints" for a question if "autoPoints", "manualPoints" or "maxAutoPoints" are specified',
             );
           }
+
           if (Array.isArray(alternative.autoPoints ?? alternative.points)) {
             errors.push(
               'Cannot specify "points" or "autoPoints" as a list for a question in a "Homework" assessment',
             );
           }
+
+          if (!courseInstanceExpired) {
+            if (
+              alternative.points !== undefined &&
+              alternative.points === 0 &&
+              alternative.maxPoints !== undefined &&
+              alternative.maxPoints > 0
+            ) {
+              warnings.push('Specifying "points": 0 when "maxPoints" > 0 is probably a mistake');
+            }
+
+            if (
+              alternative.autoPoints !== undefined &&
+              alternative.autoPoints === 0 &&
```

Comment:
Obviously this does the right thing, but isn't the `!== undefined` redundant with `=== 0`?

Reply:
Yeah, that's a reasonably point! I'll change this.

Reply:
Conditionals simplified in 30b1f1c.